static/js/giphy.js

```diff
@@ -99,6 +108,44 @@ function update_grid_with_search_term() {
     return undefined;
 }
 
+export function hide_giphy_popover() {
+    if (active_popover_element) {
+        active_popover_element.popover("hide");
+        active_popover_element = undefined;
+        return true;
+    }
+    return false;
+}
+
+function get_popover_template() {
+    let template = render_giphy_picker();
+
+    if (window.innerWidth <= 768) {
+        // Show as modal in the center for small screens.
+        template = render_giphy_picker_mobile();
+    }
+
+    return template;
+}
+
+function get_popover_placement() {
+    let placement = popovers.compute_placement(
+        active_popover_element,
+        APPROX_HEIGHT,
+        APPROX_WIDTH,
+        true,
+    );
+
+    if (placement === "viewport_center") {
+        // For legacy reasons `compute_placement` actually can
+        // return `viewport_center`, but bootstrap doesn't actually
+        // support that.
```

Comment:
We should probably explain what's going on a bit more here.

Reply:
Done.